### PR TITLE
Ts: update components

### PR DESCRIPTION
**Pull request by caro3801**
Add typescript annotations to:
- SlideUpDown
- SignUpForm
- ScaleLegend
- SharingOptions

### CODE REVIEW

File: lib/components/SharingOptionsLink.vue [hunk 1]
```diff
@@ -164,75 +181,76 @@
...
-      href () {
+      href (): string {
         return this.base + querystring.stringify(this.query)
       },
-      base () {
+      base (): string {
         return get(networks, [this.network, 'base'], '')
       },
-      args () {
+      args (): { [key: string]: string; } {
```
Comment (caro3801) [inline lib/components/SharingOptionsLink.vue:190]:
```suggestion
      args() : { [key: string]: string; } {
```

File: lib/components/ScaleLegend.vue [hunk 2]
```diff
@@ -124,7 +135,7 @@ export default {
     this.mounted = true
   },
   methods: {
-    setCursorWrapperOffset () {
+    setCursorWrapperOffset (): void {
```
Comment (caro3801) [inline lib/components/ScaleLegend.vue:138]:
```suggestion
    setCursorWrapperOffset() : void {
```

**Review by pirhoo — APPROVED** [review]

File: lib/components/ScaleLegend.vue [hunk 3]
```diff
@@ -1,9 +1,18 @@
...
-import { isFunction, isString } from 'lodash'
+<script lang="ts">
+import {isFunction, isString} from 'lodash'
 import * as d3 from 'd3'
 import * as scaleFunctions from 'd3-scale'
+import {defineComponent} from "vue";
 
-export default {
+type ClassListLegend = {'scale-legend--has-cursor':boolean}
+// eslint-disable-next-line no-unused-vars
```
[the remark this answers is not in the document]
Comment (caro3801) [reply lib/components/ScaleLegend.vue:8]: it's a typescript eslint configuration wrongly assuming that the argument of the method in the type declaration is like an unused variable. I couldn't figure it out...

[the remark this answers is not in the document]
Comment (pirhoo) [reply lib/components/ScaleLegend.vue:8]: Got it, thanks!

**Pull request merged**